fiftyone/core/odm/mixins.py

```diff
@@ -484,6 +480,11 @@ def _rename_fields(cls, sample_collection, paths, new_paths):
                 sample_collection, _paths, _new_paths
             )
 
+        # This fixes https://github.com/voxel51/fiftyone/issues/3185
+        # @todo improve list field updates in general so this isn't necessary
+        if schema_paths:
```

Comment:
Previously `_reload_fields()` was called earlier, before issuing potentially long-running aggregations to rename the fields in the database. Now the reload happens later, immediately before the dataset doc's schema is updated, which should reduce potential errors due to concurrent dataset edits.